### DISCUSSION TITLE
Login not working on virtual hosts

### DISCUSSION THREAD

**Discussion by vimkaf** [codeigniter4/shield#962] [Q&A]
I have an app running on CI v4.4.3 and Shield v1.0.0.0-beta.7 and PHP v8.2

Whenever i run php spark serve all works fine 

but when i use the virtual host appname.local the login functionality is not working

The virtual host is created by Laragon.

**Comment by datamweb** [comment]
Your explanation is not enough. We don't even know what error you are seeing.

In any case, I don't think the problem is with the Shield, you should look at the base_url and ... do it correctly.

**Reply by vimkaf** [reply]
I am not getting any error. i just get redirected back to the login page but if i enter a wrong email and password i get the error message. A valid credential does not log me in.

**Reply by vimkaf** [reply]
![image](https://github.com/codeigniter4/shield/assets/2923113/ec45f90e-a070-423f-8908-7d783f557154)

**Reply by kenjis** [reply]
Don't use screenshot for code. We cannot copy and paste easily.

**Reply by kenjis** [reply]
Comment out the line `app_baseURL`.

**Comment by kenjis** [comment]
> Whenever i run php spark serve all works fine
> but when i use the virtual host appname.local the login functionality is not working

This shows the Laragon configuration is something wrong.

**Reply by vimkaf** [reply]
I have checked the error logs in laragon but nothing..

This is the conf file for the virtual host

![image](https://github.com/codeigniter4/shield/assets/2923113/66742ba6-d5e6-4066-9d06-4337c266f855)

**Reply by kenjis** [reply]
Show what happened exactly. "Not working" does not tell anything to us.
You may not know this, but we cannot see your screen.

**Comment by datamweb** [comment]
**CI v4.4.3 and Shield v1.0.0.-beta.7 in Laragon**

![Screenshot 2023-11-26 081920](https://github.com/codeigniter4/shield/assets/9530214/c8a1c702-4198-4091-945f-53f8418006cf)

**Comment by vimkaf** [comment]
It worked without SSL enabled but when you enable SSL in laragon ( Menu > Apache > SSL) it doesn't work. 

What I mean by it worked is that the credentials are validated and the login redirect action is also ok but as you can see from the screenshot below Shield doesn't log in the user.

![image](https://github.com/codeigniter4/shield/assets/2923113/839d1007-fcbd-4c68-bad3-a7241f77aee7)

**Reply by datamweb** [reply]
Unfortunately, I can't reproduce your problem.

![Screenshot 2023-11-27 011659](https://github.com/codeigniter4/shield/assets/9530214/c281a3aa-3bf3-4470-b4b2-aa0a009edce8)
